olca-app/src/org/openlca/app/editors/graphical/model/ExchangeItem.java

```diff
@@ -139,13 +143,15 @@ private static int typeOrderOf(Exchange e) {
 		};
 	}
 
-	public static int compare(ExchangeItem e1, ExchangeItem e2) {
-		if (e1 == null && e2 == null)
+	public static int compare(ExchangeItem e1, ExchangeItem e2, Flow refFlow) {
+		if (e1 == null || e2 == null)
 			return 0;
-		if (e1 != null && e2 == null)
+		if (e1.exchange == null || e2.exchange == null)
+			return 0;
+		if (e1.exchange.flow != null && e1.exchange.flow.equals(refFlow))
```

Comment:
This would move the reference flow at the end of the list but it should be at the top of the list, returning -1 instead. Also, the reference flow is not the same as the quantitative reference of the process. In rare cases you could have the reference flow on the input and output side of a process (well, it is quite common in IO databases), then it would be also at the bottom/top of the input list. However, the initial request was more for the exchange tables in processes. For the graph editor it is not really necessary as the large elementary flow lists are not displayed by default here.

Reply:
I see your point. Of course, I meant reference flow. It is a quantitative reference exchange only for processes.
I'll delete that branch.

Reply:
No, the point was the parameter is of type Flow, but the quantitative reference is of type Exchange.

Reply:
Ah yes, I see!
I should have compared `exchangeItem.exchange` with `process.quantitativeReference` with `PROCESS`, `productSystem.referenceExchange` with `PRODUCT_SYSTEM` and  `exchangeItem.exchange.flow` with `result.referenceFlow.flow` with `RESULT`.
That's what you meant?

Reply:
yes, exactly like this